synapse/dendrite/coordinator.py

```diff
@@ -0,0 +1,261 @@
+    completed status. There can be multiple Parsers online for a given queue, as they will all work independently to
+    drain the queue without duplicating work.
+
+    Keyword Arguments:
+        svcbus (str):
+            URL for the SvcBus service that is used to locate the Cortex, Axon, and Jobs services.
+
+        core (str):
+            The name that the Cortex is shared as on the SvcBus. The Cortex must be shared with a 'link'
+            property that is set with the URL that the Cortex is listening on. This is needed so that the
+            Coordinator can connect directly to the Cortex for listening to the tufo change events.
+
+        axon (str):
+            The name that the Axon is shared as on the SvcBus. The Axon must be shared with a 'link'
+            property that is set with the URL that the Cortex is listening on. This is needed so that the
+            Coordinator can connect directly to the Axon for looking up and reading axon nodes.
+
+        jobs (str):
+            The name that the Jobs service is shared as on the SvcBus.
+
+        mintime (int):
+            Timestamp represented as milliseconds since the epoch. When specified, all file:bytes tufos
+            created since this time will be processed on startup. This is useful for ensuring that any
+            work that would have been processed during Coordinator downtime is accounted for.
+
+    Raises:
+        TypeError:
+            If one of 'svcbus', 'core', 'axon', 'jobs' kwargs are not provided.
+
+        synapse.exc.NoSuchObj:
+            If the Coordinator is unable to connect to the Cortex, Axon, or Jobs services.
+
+    '''
+    _REQUIRED_ARGS = set(('svcbus', 'core', 'axon', 'jobs'))
+    _BYTES_TO_READ = 1024
+
+    def __init__(self, **kwargs):
+        self.opts = kwargs
+        self._assertOpts()
+        self.svcbus = s_service.openurl(self.opts.get('svcbus'))
+        coreSvc = self.svcbus.getSynSvcByName(self.opts.get('core'))
+        self.listenCore = s_telepath.openurl(coreSvc[1].get('link'))
+        axonSvc = self.svcbus.getSynSvcByName(self.opts.get('axon'))
+        self.axonLink = axonSvc[1].get('link')
+        self.axon = s_telepath.openurl(self.axonLink)
+        self.jobs = self.svcbus.getNameProxy(self.opts.get('jobs'))
+        self._assertConnected()
+        self._registerListeners()
+        self._initCortex()
+        self._initState()
+        self._processExistingTufos()
+
+    def register(self, mimeType, queue):
+        '''
+        Registers a MIME type with a queue name.
+
+        This method is intended to be called remotely via Telepath by the individual Parsers. When a Parser comes
+        online, it should call this method to register the MIME type that it is interested in Parsing. The queue name
+        can be any value, however, care should be taken that queue names do not collide unintentionally in a given
+        deployment.
+
```

Comment:
Right, there could potentially be many different parser implementations for a given mime type. The idea is that a given Parser would register a specific queue name to consume off of, and multiple instances of that Parser would probably want to consume off of the same named queue. Letting it be a 'people friendly' made sense to me for config specification, deployment, etc.